### PR TITLE
Simplified component message hierarchy

### PR DESCRIPTION
**Pull request by prprabhu-ms**
Replaces generic type hierarchy for messages in the UI components with discriminated unions.

The new type hierarchy is easier to understand and loses none of the concrete type safety.

### CODE REVIEW

**Comment by prprabhu-ms** [comment]
I found the message hierarchy quite confusing. PoC to see if this form is simpler and to see if I missed someething.

File: packages/react-components/src/types/ChatMessage.ts [hunk 1]
```diff
@@ -21,129 +21,124 @@ export type MessageAttachedStatus = 'bottom' | 'top';
 export type MessageContentType = 'text' | 'html' | 'richtext/html' | 'unknown';
```
Comment (prprabhu-ms) [inline packages/react-components/src/types/ChatMessage.ts:21]: This file has the key type rework. Everything else is a fairly straightforward porting over to the new types.

**Review by JamesBurnside — APPROVED** [review]
Nice!

File: packages/chat-component-bindings/src/messageThreadSelector.ts [hunk 2]
```diff
@@ -40,56 +46,50 @@ const convertToUiChatMessage = (
...
-              (participant): CommunicationParticipant => ({
-                userId: toFlatCommunicationIdentifier(participant.id),
-                displayName: participant.displayName
-              })
-            ) ?? [],
-        type: systemMessageType,
-        messageId: message.id,
-        iconName: systemMessageType === 'participantAdded' ? 'PeopleAdd' : 'PeopleBlock'
-      }
+      messageType: 'system',
```
Comment (anjulgarg) [inline packages/chat-component-bindings/src/messageThreadSelector.ts:67]: any reason for moving away from message.type to message.messageType?
`message` prefix feels redundant.

Comment (prprabhu-ms) [reply packages/chat-component-bindings/src/messageThreadSelector.ts:67]: To disambiguate from these other fields: `systemMessageType` (which further discriminates `SystemMessage` variants) and `contentType` (a field in `ChatMessage`).

Comment (anjulgarg) [reply packages/chat-component-bindings/src/messageThreadSelector.ts:67]: Approved :)

**Comment by prprabhu-ms** [comment]
Manually sanity checked ChatComposite via storybook.

Checked artifacts from the failed UI test -- the checked in snapshot for the localization test is clearly wrong for linux (shows an error bar).

**Review by alkwa-msft — APPROVED** [review]

**Pull request merged**